boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk_defconfig

```diff
@@ -25,3 +25,9 @@ CONFIG_CLOCK_CONTROL=y
 
 # For Misc Register Map
 CONFIG_SYSCON=y
+
+# Misc
+CONFIG_HEAP_MEM_POOL_SIZE=16384
+
+# Enable QSPI Flash
+CONFIG_FLASH=y
```

Comment:
is this required to even run minimal samples? `_defconfig` should only enable minimal settings

Reply:
Hi @gmarull,

We are covering the case that user might write their own sample, whenever they build with our board on their sample we need to make sure that the flash driver is build together with their sample, thus, for all the driver that we are enabling it will be added to the board defconfig, including the CONFIG_FLASH.

Reply:
Not really samples, we are trying to release driver that is fully enabled at board level.

Reply:
Please checl https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#general-recommendations `_defconfig` needs to enable minimal peripherals, it's application responsibility to enable what it needs. This has been the rule for all in-tree boards.

Reply:
In near future, we would upstream a full feature of a remote system upgrade, which require flash to be enabled per our implementation, enabling flash at application eg: at flash_shell only the remote system upgrade may not work.